openassessment/xblock/test/test_submission.py

```diff
@@ -199,6 +199,29 @@ def test_open_submitted(self, xblock):
             }
         )
 
+    @scenario('data/submission_open.xml', user_id="Bob")
+    def test_cancelled_submission(self, xblock):
+        submission = xblock.create_submission(
+            xblock.get_student_item_dict(),
+            'A man must have a code'
+        )
+        xblock.get_workflow_info = Mock(return_value={
+            'status': 'cancelled',
+            'submission_uuid': submission['uuid']
+        })
+        self._assert_path_and_context(
+            xblock, 'openassessmentblock/response/oa_response_cancelled.html',
+            {
+                'submission_due': dt.datetime(2999, 5, 6).replace(tzinfo=pytz.utc),
+                'student_submission': submission,
+                'allow_file_upload': False,
+                'has_peer': True,
+                'has_self': True,
+                'allow_latex': False,
+                'workflow_cancellation': None
```

Comment:
Well i'm checking based on the workflow status which is 'cancelled' and ignoring the further information but i think you are right it could be confusing. i will add the 'workflow_cancellation' information to make it more readable.